Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp

```diff
@@ -123,6 +123,15 @@ template<typename CharacterType> static inline bool isCharAfterUnquotedAttribute
     return character == ' ' || character == '>' || isHTMLSpace(character);
 }
 
+template<typename T> static bool insertInUniquedSortedVector(Vector<T>& vector, const T& value)
+{
+    auto it = std::lower_bound(vector.begin(), vector.end(), value);
```

Comment:
I am guessing that using binary search to insert into a kept-sorted vector is the best way to go, but I can’t help wondering if there is a better data structure and algorithm for checking for duplicates. For example, when the vector is very short, a linear search will be faster than a binary search, but of course then it won’t be sorted for when the vector gets longer.

Reply:
I tried both linear search (unsorted vector) and binary search with a sorted vector. They were both perf neutral on Speedometer (unlike the HashSet which was a regression). I went with the sorted vector / binary search because it should be faster with larger vectors and there was no cost on Speedometer.